src/dbnode/persist/fs/types.go

```diff
@@ -172,6 +172,9 @@ type DataFileSetReader interface {
 
 	// MetadataRead returns the position of metadata read into the volume
 	MetadataRead() int
+
+	// OrderedByIndex returns true if the reader reads the data in the order of index.
```

Comment:
nit: should add some comment on default order if this is not enabled eg `If this is false, data is ordered based on their offsets within the data file` or something like that

Reply:
👍